clients/venice-client/src/main/java/com/linkedin/venice/fastclient/RetriableAvroGenericStoreClient.java

```diff
@@ -317,13 +319,12 @@ private <R extends MultiKeyRequestContext<K, V>, RESPONSE> void retryStreamingMu
         if (throwable != null || multiKeyLongTailRetryManager == null
             || multiKeyLongTailRetryManager.isRetryAllowed(pendingKeysFuture.keySet().size())) {
           Set<K> pendingKeys = Collections.unmodifiableSet(pendingKeysFuture.keySet());
+          // Prepare the retry context and track excluded routes on a per-partition basis
           R retryRequestContext =
               requestContextConstructor.construct(pendingKeys.size(), requestContext.isPartialSuccessAllowed);
 
           requestContext.retryContext.retryRequestContext = retryRequestContext;
           LOGGER.debug("Retrying {} incomplete keys", retryRequestContext.numKeysInRequest);
-          // Prepare the retry context and track excluded routes on a per-partition basis
-          retryRequestContext.setRoutesForPartitionMapping(requestContext.getRoutesForPartitionMapping());
```

Comment:
Is `setRoutesForPartitionMapping() `from original to `retryRequestContext` intended to be removed? How does the retry now knows which replicas to ignore as already used?

Reply:
Good catch!
Will add a small unit test.